src/costmap_to_polygons.cpp

```diff
@@ -37,12 +37,71 @@
  *********************************************************************/
 
 #include <costmap_converter/costmap_to_polygons.h>
+#include <costmap_converter/misc.h>
 #include <boost/thread.hpp>
 #include <boost/thread/mutex.hpp>
 #include <pluginlib/class_list_macros.h>
 
 PLUGINLIB_EXPORT_CLASS(costmap_converter::CostmapToPolygonsDBSMCCH, costmap_converter::BaseCostmapToPolygons)
 
+namespace
+{
+
+/**
+ * @brief Douglas-Peucker Algorithm for fitting lines into ordered set of points
+ * 
+ * Douglas-Peucker Algorithm, see https://en.wikipedia.org/wiki/Ramer%E2%80%93Douglas%E2%80%93Peucker_algorithm
+ * 
+ * @param pointList the initial set of points
+ * @param epsilon distance criteria for removing point if it is smaller to the line segment than this
+ * @param result the simplified polygon
+ */
+void douglasPeucker(const std::vector<geometry_msgs::Point32> &pointList, double epsilon,
```

Comment:
why not just returning the result from the function? then you can skip the result.clear() and simplify the lines 90-96. I also would suggest to change the signature std::vector<Point32> douglasPeucker(ConstIter first, ConstIter last, double epsilon) so we don't allocate new data everytime we call recursion

Reply:
good idea!